crates/ui/src/components/icon.rs

```diff
@@ -496,12 +496,16 @@ impl Icon {
 
 impl RenderOnce for Icon {
     fn render(self, cx: &mut WindowContext) -> impl IntoElement {
-        svg()
+        let svg = svg()
```

Comment:
We don't want to make any changes to the `Icon` yet, as right now the intent in Zed is for all icons to by monochrome.

Reply:
This PR doesn't change the icons. It just allows the renderer to work correctly and respect the colours. All the default icons are still monochrome.

Reply:
But it's still making changes to the `Icon` component, which could have unintended effects.

I would prefer if we kept the scope limited just to GPUI right now.

Reply:
Well, I am not sure that the current defaults are good. Currently, GPUI makes all SVGs monochrome. This is a huge surprise to anyone who has worked with a  UI framework. 

This PR enables the correct behaviour using `with_polychrome`, but if I was going to write this from scratch, I would reverse the default. Respect the colours, unless explicitly requested by the user.

So, if you prefer, I can do this, and explicitly force all Icons to be monochrome until we decide what should be done regarding the icons.

Reply:
So like... maybe I'm naive, but why does the framework need to override the colors? Why can't the SVG itself just be monochrome? Sounds to me like a utility abstraction that is doing more harm than good based on how much the community seems to _really_ want colorful icons.